packages/range.yaml

```diff
@@ -240,14 +250,19 @@ mqtt:
       unique_id: user_temperature_offset
       command_topic: "geappliances/oven/erd/0x5106/write"
       command_template: >
-        {{ '%02x' % (value | int) }}
+        {{ '%02x' % value|int(0) }}
       state_topic: "geappliances/oven/erd/0x5106/value"
       value_template: >
-        {{ value|int }}
+        {% set output = (value|int(base=16)) %} 
+        {%- if output > 32767 -%} 
+            {{ (output - 65535) }} 
```

Comment:
```suggestion
        {%- if output > 127 -%} 
            {{ (output - 255) }} 
```

This is an 8-bit signed value instead of a 16-bit signed value so this needs to be modified a bit

Reply:
It's actually interesting, I was testing it using mqtt explorer setting the values directly and it seems to have no issue going above 127 and retaining the value.

Reply:
I double-checked and our docs show this as being 8-bit so I'm confused how that could be possible.

Reply:
tested this and if I put in a value below 0 it changes to `-46.0`

Reply:
In MQTT explorer how many digits or bytes is the value for this ERD? The docs say it should be 8 bits (2 hex digits).

Reply:
Okay, then something is definitely off. For the `32767` and `65535` values to be correct, it would need to be 4 hex digits. Can you go into the settings menu on your range and set a negative offset then see what raw value shows up in MQTT explorer? With that info I should be able to work out what the mapping is.

Reply:
For `-1`
<img width="208" alt="image" src="https://github.com/user-attachments/assets/c424e34a-8198-4081-bd08-385323231d93" />
 and `85`
<img width="218" alt="image" src="https://github.com/user-attachments/assets/74313bcb-64d2-49a7-858b-91faf2a4546e" />

My yaml is set to `{{ '%02x' % value|int(0) }}`. If I change it to `'%03x'` it still works and if I do `-180` I get:
<img width="221" alt="image" src="https://github.com/user-attachments/assets/b8dab66a-109a-45a2-9a5c-798bbdc7026c" />

Reply:
You set those values via MQTT Explorer, right? Can you set them via the UI on your range? That's what I want to see so that I can understand how the range interprets them.

Reply:
On the range it would be in the "special features menu" and you probably need to hold multiple keys to enter this menu. I think that for your oven you need to hold Bake and Broil.

What this setting does is set a temperature offset for your cavity. For example, if you find that it tends to cook warmer than it should you can set a negative temperature offset so that it cooks at a lower temperature at a given setting.